File: widgetssdk/src/main/java/com/glia/widgets/core/notification/domain/ShowVideoCallNotificationUseCase.java
package com.glia.widgets.core.notification.domain;

import com.glia.widgets.core.engagement.domain.IsCallVisualizerUseCase;
import com.glia.widgets.core.notification.device.INotificationManager;

public class ShowVideoCallNotificationUseCase {
    private final INotificationManager notificationManager;
    private final IsCallVisualizerUseCase isCallVisualizerUseCase;

    public ShowVideoCallNotificationUseCase(
            INotificationManager notificationManager,
            IsCallVisualizerUseCase isCallVisualizerUseCase
    ) {
        this.notificationManager = notificationManager;
        this.isCallVisualizerUseCase = isCallVisualizerUseCase;
    }

    public void execute() {
        notificationManager.showVideoCallNotification(isCallVisualizerUseCase.execute());
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/notification/domain/ShowVideoCallNotificationUseCase.kt
package com.glia.widgets.core.notification.domain

import com.glia.widgets.core.callvisualizer.domain.IsCallVisualizerUseCase
import com.glia.widgets.core.notification.device.INotificationManager

internal class ShowVideoCallNotificationUseCase(
    private val notificationManager: INotificationManager,
    private val isCallVisualizerUseCase: IsCallVisualizerUseCase
) {
    operator fun invoke() = notificationManager.showVideoCallNotification(isCallVisualizerUseCase())
}